src/lib/sync_handler/sync_handler.ml

```diff
@@ -30,10 +35,15 @@ module Make (Inputs : Inputs_intf) :
       Sequence.of_lazy
         (lazy (Sequence.of_list @@ get_breadcrumb_ledgers frontier))
     in
-    Sequence.append
-      (Sequence.singleton
-         (Transition_frontier.shallow_copy_root_snarked_ledger frontier))
-      ledger_breadcrumbs
+    let root_ledger =
+      Ledger.Any_ledger.cast
+        (module Ledger.Db)
+        (Transition_frontier.root_snarked_ledger frontier)
+    in
+    let mask =
+      Ledger.Maskable.register_mask root_ledger (Ledger.Mask.create ())
```

Comment:
Would we ever unregister this mask? Would this be a trouble? I saw this function is called by `answer_query`. I think every time `answer_query` is called, a new mask is created here.

Reply:
Looks like a leak to me.

Reply:
This `get_ledger_by_hash` function is called by `Sync_handler.answer_query`. And this is only called when we handle requests for bootstrap. So `root_snarked_ledger` should be the only one we care about. Maybe we don't need `get_ledger_by_hash` function at all.

Reply:
I addressed this leak in this commit https://github.com/CodaProtocol/coda/pull/3439/commits/34b325bbe804eb1a9ed01145f4566ff5ecc10fb4